packages/shared/src/lib/core/activity/utils/evm/generateSmartContractActivity.ts

```diff
@@ -0,0 +1,29 @@
+import { BASE_TOKEN_ID } from '@core/token'
+import { ActivityType } from '../../enums'
+import { PersistedEvmTransaction, SmartContractActivity } from '../../types'
+import { IChain } from '@core/network'
+import { generateBaseEvmActivity } from './generateBaseEvmActivity'
+import { IAccountState } from '@core/account/interfaces'
+
+export async function generateSmartContractActivity(
+    transaction: PersistedEvmTransaction,
+    chain: IChain,
+    account: IAccountState
+): Promise<SmartContractActivity> {
+    const baseEvmActivity = await generateBaseEvmActivity(transaction, chain, transaction.to?.toString(), account)
+
+    const baseTokenTransfer = {
+        tokenId: BASE_TOKEN_ID,
+        rawAmount: String(transaction.value) ?? '0',
```

Comment:
This needs rewriting when the BigInt branch has been merged. Let's wait with merging this until that one is in.